resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/YarnSparkHadoopUtil.scala

```diff
@@ -196,4 +197,20 @@ object YarnSparkHadoopUtil {
       0
     }
   }
+
+  /**
+   * Get offHeap memory size from [[ExecutorResourceRequest]]
+   * return 0 if MEMORY_OFFHEAP_ENABLED is false.
+   */
+  def executorOffHeapMemorySizeAsMb(sparkConf: SparkConf,
```

Comment:
these functions are almost identical. Lets create a helper function checkOffHeapEnabled that has the core logic and just pass in size we want...

def executorOffHeapMemorySizeAsMb(sparkConf: SparkConf,  execRequest: ExecutorResourceRequest): Long = {
   checkOffHeapEnabled(sparkConf, execRequest.amount).toLong

The version above would call with checkOffHeapEnabled(sparkConf, Utils.memoryStringToMb(sparkConf.get(MEMORY_OFFHEAP_SIZE).toString))